src/app/rosetta/network.ml

```diff
@@ -0,0 +1,88 @@
+open Core_kernel
+open Async
+open Models
+
+module Get_status =
+[%graphql
+{|
+  query {
+    bestChain {
+      stateHash
+      protocolState {
+        blockchainState {
+          utcDate @bsDecoder(fn: "Int64.of_string")
+        }
+        consensusState {
+          blockHeight @bsDecoder(fn: "Graphql.Decoders.int64")
+        }
+      }
+    }
+    daemonStatus {
+      peers
+    }
+  }
+|}]
+
+module Get_version = [%graphql {|
+  query {
+    version
+  }
+|}]
+
+(* TODO: Is there a better way to get the genesis block info *)
+let genesis_block_query =
+  Caqti_request.find Caqti_type.unit Caqti_type.string
+    "SELECT state_hash FROM blocks WHERE height = 0"
+
+let router ~graphql_uri (module Db : Caqti_async.CONNECTION) route body =
+  let open Async.Deferred.Result.Let_syntax in
+  match route with
+  | ["list"] ->
+      let%map _meta = Mappers.parse @@ Metadata_request.of_yojson body in
+      Network_list_response.to_yojson
+        { Network_list_response.network_identifiers=
+            [ { Network_identifier.blockchain= "coda"
+              ; network= "testnet"
+              ; sub_network_identifier= None } ] }
+  | ["status"] ->
+      (* TODO: Check that the network corresponds to the node we're connected to *)
+      let%bind _network = Mappers.parse @@ Network_request.of_yojson body in
+      let%bind res = Graphql.query (Get_status.make ()) graphql_uri in
+      let%bind latest_block =
+        Deferred.return
+          ( match res#bestChain with
```

Comment:
should we be getting this block from the Db long term? Or do you think we should always cache the latest block via graphql or something.

Reply:
yeah probably want to get it from the db